lib/common/common/Int64Math.cpp

```diff
@@ -30,6 +37,16 @@ Int64Math::Mul(int64 left, int64 right, int64 *pResult)
     return (left != 0 && right != 0 && (*pResult / left) != right);
 #endif
 }
+#else
+// FIXME: this probably needs checks for __int128 at least.
+bool
+Int64Math::Mul(int64 left, int64 right, int64 *pResult)
+{
+    __int128 result = (__int128)left * (__int128)right;
+    *pResult = (int64)result;
```

Comment:
Why promote them both for the multiplication only to throw away the extra precision in a cast in the very next line? Was there a warning here you were trying to avoid?